File: 프로그래머스/lv2/12953. N개의 최소공배수/N개의 최소공배수.py
def gcd(a, b):
    while b > 0:
        a, b = b, a % b
    return a

def lcm(a, b):
    return a * b // gcd(a, b)

def solution(arr):
    arr.sort()
    while len(arr) > 1:
        a, b = arr.pop(), arr.pop()
        arr.append(lcm(a, b))
    return arr[0]

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
import java.util.*;

class Solution {
    public int solution(int[] arr) {
        
        return getLcm(arr);
    }
    
    // 최소 공배수 = 모든 값의 곱 / 최대공약수
    public int getLcm(int[] arr) {
        if (arr.length == 1) {
            return arr[0];
        }
        
        int gcd = getGcd(arr[0], arr[1]);
        int lcm = (arr[0] * arr[1]) / gcd;
        
        for(int i=2; i<arr.length; i++) {
            gcd = getGcd(lcm, arr[i]);
            lcm = (lcm * arr[i]) / gcd;
        }
        
        return lcm;
    }
    
    // 재귀함수를 통해서 최대공약수 구하기
    public int getGcd(int num1, int num2) {
        if (num1 % num2 == 0) {
            return num2;
        }
        
        return getGcd(num2, num1 % num2);
    }
}